### PR TITLE
fix(network): refining the connection limit

### PR DESCRIPTION
**Pull request by b00f**
## Description

This PR includes the following changes:

- Refines the maximum connection limit.
- Eliminates the need for users to define a minimum connection limit.
- Normalizes and logarithmically scales the maximum connection limit for improved results.

### CODE REVIEW

**Comment by b00f** [comment]
Discussion about the connection limit can be found here: https://github.com/libp2p/go-libp2p/issues/2616

File: network/peermgr.go [hunk 1]
```diff
@@ -39,8 +39,8 @@ func newPeerMgr(ctx context.Context, h lp2phost.Host,
 	b := &peerMgr{
 		ctx:            ctx,
 		bootstrapAddrs: conf.BootstrapAddrInfos(),
-		minConns:       conf.MinConns,
-		maxConns:       conf.MaxConns,
+		minConns:       conf.ScaledMaxConns(),
```
Comment (kehiy) [inline network/peermgr.go:42]:
```suggestion
		minConns:       conf.ScaledMinConns(),
```

@b00f I think it shuld be min.

File: network/peermgr.go [hunk 2]
```diff
@@ -39,8 +39,8 @@ func newPeerMgr(ctx context.Context, h lp2phost.Host,
...
+		maxConns:       conf.ScaledMinConns(),
```
Comment (kehiy) [inline network/peermgr.go:43]:
```suggestion
		maxConns:       conf.ScaledMaxConns(),
```

and max for this one.

**Review by kehiy — CHANGES_REQUESTED** [review]

**Review by kehiy — APPROVED** [review]
@b00f looks good to me.

**Review by amirvalhalla — CHANGES_REQUESTED** [review]
@b00f  please resolve requested changes.

File: network/utils.go [hunk 3]
```diff
@@ -131,36 +130,30 @@ func SubnetsToFilters(subnets []*net.IPNet, action multiaddr.Action) *multiaddr.
...
-	limit.TransientLimitIncrease.Streams = util.LogScale(minConns)
-
-	return limit
+func BuildConcreteLimitConfig(maxConns int) lp2prcmgr.ConcreteLimitConfig {
+	changes := lp2prcmgr.PartialLimitConfig{}
+
+	updateResourceLimits := func(limit *lp2prcmgr.ResourceLimits, maxConns, coefficient int) {
+		maxConnVal := lp2prcmgr.LimitVal(maxConns * coefficient)
+
+		limit.ConnsOutbound = maxConnVal / 2
```
[the remark this answers is not in the document]
Comment (b00f) [reply network/utils.go:139]: I think yes, check the default limits in LibP2P, they put the same value for in-bound and out-bound connections.

**Review by amirvalhalla — APPROVED** [review]

**Comment by themantre** [comment]
> @b00f please resolve requested changes.

It is done. Please check

**Pull request merged**